sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ManifestFileCommitProtocol.scala

```diff
@@ -114,7 +117,10 @@ class ManifestFileCommitProtocol(jobId: String, path: String)
   }
 
   override def abortTask(taskContext: TaskAttemptContext): Unit = {
-    // Do nothing
-    // TODO: we can also try delete the addedFiles as a best-effort cleanup.
+    // best effort cleanup of incomplete files
+    if (addedFiles.nonEmpty) {
+      val fs = new Path(addedFiles.head).getFileSystem(taskContext.getConfiguration)
+      addedFiles.foreach { file => fs.delete(new Path(file), false) }
```

Comment:
handle IOException here?

Reply:
I just followed how HadoopMapReduceCommitProtocol has been doing, but if we want to try deleting all files with ignoring exception, we can apply it to both places: HadoopMapReduceCommitProtocol and here.

@zsxwing What do you think?

Reply:
I'm okey with the current codes. If one IOException was thrown, the following delete requests would probably fail since this is the same file system.

Reply:
If the exception is propagated up the task/job may fail?

Reply:
This method is to abort a failed task.

Reply:
okay, the `IOException` would be suppressed and original cause is propagated so it seems fine.

Reply:
> If one IOException was thrown, the following delete requests would probably fail since this is the same file system.

I've seen some cases intermittent problems but I'm also OK with not handling exception since majority of the cases it's permanent issue.